tests/Filter/Model/FilterDataTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Filter\Model;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\AdminBundle\Filter\Model\FilterData;
+
+final class FilterDataTest extends TestCase
+{
+    /**
+     * @dataProvider getInvalidTypes
+     *
+     * @param mixed $type
+     */
+    public function testTypeMustBeNumericOrNull($type): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('The "type" value MUST be "null" or "int", string given.');
+
+        FilterData::fromArray(['type' => 'string']);
```

Comment:
The expected message should be also updated.
```suggestion
        FilterData::fromArray(['type' => $type]);
```